src/rust/zh.dmzj/src/lib.rs

```diff
@@ -0,0 +1,414 @@
+// v3api now shutdown the functionality to fetch manga detail and chapter list,
+// so move these logic to v4api
+const V4_API_URL: &str = "https://nnv4api.muwai.com"; // https://v4api.dmzj1.com
+const API_URL: &str = "https://api.dmzj.com";
+const API_PAGELIST_OLD_URL: &str = "https://api.m.dmzj.com";
+const API_PAGELIST_WEBVIEW_URL: &str = "https://m.dmzj.com/chapinfo";
+const IMAGE_URL: &str = "https://images.dmzj.com";
+const IMAGE_SMALL_URL: &str = "https://imgsmall.dmzj.com";
+
+const FILTER_GENRE: [i32; 42] = [
+    0, 4, 3243, 3242, 17, 3244, 3245, 3249, 3248, 3246, 16, 14, 7, 6, 5, 8, 9, 13, 12, 11, 10,
+    3250, 3251, 5806, 5345, 5077, 5848, 6316, 7900, 7568, 6437, 4518, 4459, 3254, 3253, 3252, 3255,
+    6219, 3328, 3365, 3326, 3325,
+];
+
+const FILTER_STATUS: [i32; 3] = [0, 2309, 2310];
+const FILTER_READER: [i32; 4] = [0, 3262, 3263, 3264];
+const FILTER_TYPE: [i32; 7] = [0, 2304, 2305, 2306, 2307, 2308, 8453];
+
+#[get_manga_list]
+pub fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+    let mut is_keyword: bool = false;
+    let mut keyword: String = String::new();
+    let mut sort: i32 = 0;
+    let mut filters_list: Vec<i32> = Vec::new();
+
+    for filter in filters {
+        match filter.kind {
+            FilterType::Title => {
+                let title = filter.value.as_string()?.read();
+                is_keyword = !title.is_empty();
+
+                if is_keyword {
+                    keyword = title.clone();
+                    break;
+                }
+            }
+            FilterType::Select => {
+                let index = filter.value.as_int()? as usize;
+                let id = match filter.name.as_str() {
+                    "连载状态" => FILTER_STATUS[index],
+                    "读者" => FILTER_READER[index],
+                    "地区" => FILTER_TYPE[index],
+                    "分类" => FILTER_GENRE[index],
+                    _ => continue,
+                };
+
+                if id != 0 {
+                    filters_list.push(id);
+                }
+            }
+            FilterType::Sort => {
+                let value = match filter.value.as_object() {
+                    Ok(value) => value,
+                    Err(_) => continue,
+                };
+                sort = value.get("index").as_int()? as i32;
+            }
+            _ => continue,
+        }
+    }
+
+    let mut manga_arr: Vec<Manga> = Vec::new();
+
+    if is_keyword {
+        let url = format!(
+            "http://s.acg.dmzj.com/comicsum/search.php?s={}",
+            &helper::encodeURI(&keyword)
+        );
+
+        let data = {
+            let req = helper::GET(url);
+            let r = req.string();
+
+            let r = r
+                .strip_prefix("var g_search_data = ")
+                .unwrap()
+                .strip_suffix(";")
+                .unwrap();
+
+            json::parse(r.as_bytes().clone()).as_array()?
```

Comment:
You probably don't need to `.clone()` this
```suggestion
            json::parse(r.as_bytes()).as_array()?
```

Reply:
Thanks a lot!